pulp_ansible/app/tasks/collections.py

```diff
@@ -503,6 +512,27 @@ async def run(self):
         """
         tasks = []
         loop = asyncio.get_event_loop()
+        root, api_version = await self._get_collection_api_root(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                pass
+
+            collection_info = self.collection_info or []
+            sources = {r.source for r in collection_info if r.source}
+            sources.add(self.remote.url)
+
+            published = metadata.get("published")
+            if published == self.remote.last_synced_metadata_time and len(sources) < 2:
+                log.info(_("no-op"))
+                return
```

Comment:
we may have `self.remote.url` and `sources` pointing to different repos, should we only consider `self.remote.url` for no-op?

Reply:
Yeah maybe we should leave off the inspection of sources from this PR at least to keep it simple from the start... wdyt?